src/services/addItemToFirebase.tsx

```diff
@@ -1,17 +1,35 @@
 import { db } from '../firebase';
 import { doc, updateDoc, arrayUnion } from 'firebase/firestore';
+import { getAuth } from 'firebase/auth';
 
-export const addItemToFirebase = async (item, category) => {
+interface Item {
+  id: string;
+  name: string;
+  [key: string]: any;
+}
+
+export const addItemToFirebase = async (
+  item: Item,
+  category: string,
+): Promise<string> => {
   try {
-    const userRef = doc(db, 'users', 'userId'); // Replace 'userId' with the actual user ID
-    const categoryRef = doc(userRef, category); // Reference to the category document (weapons, spells, equipment)
+    const auth = getAuth();
+    const currentUser = auth.currentUser;
+
+    if (!currentUser) {
+      throw new Error('User not authenticated');
+    }
```

Comment:
Tak jak pisalem w poprzednim komentarzu, jeżeli to sprawdzamy to zakładam żę spodziewamy się niezalogowaneog usera? Jeżeli tak, to mozemy to zrobić po stronie funkcji, jednorazowo. Jezeli nie, to user nie powinien miec wgl mozliwosci wykonania tej akcji lub wejscia na ta strone nawet.